From 13d6798ce16fa57a8130416d02f263cca4a26a2c Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Thu, 23 Feb 2017 08:18:36 +0000
Subject: [PATCH] refpolicy: fix avc denials for dhcpc_t

Reference sources: selinux-policy-3.13.1-225.6.fc25.src.rpm
Changes come from: policy-f25-base.patch

* allow dhcpc_t search /run/systemd/journal
* allow dhcpc_t search /etc/ssl

Fix avc denials:

  avc: denied { search } for pid=973 comm="dhclient" \
  name="journal" dev="tmpfs" ino=12118 \
  scontext=system_u:system_r:dhcpc_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:syslogd_var_run_t:s15:c0.c1023 tclass=dir

  avc: denied { search } for pid=983 comm="dhclient" \
  name="ssl" dev="sda1" ino=1358436 \
  scontext=system_u:system_r:dhcpc_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:cert_t:s0 tclass=dir

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/system/sysnetwork.te | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/policy/modules/system/sysnetwork.te b/policy/modules/system/sysnetwork.te
index b144d52..45dd5f2 100644
--- a/policy/modules/system/sysnetwork.te
+++ b/policy/modules/system/sysnetwork.te
@@ -131,6 +131,7 @@ files_search_var_lib(dhcpc_t)
 files_dontaudit_search_locks(dhcpc_t)
 files_getattr_generic_locks(dhcpc_t)
 files_manage_var_files(dhcpc_t)
+mls_file_read_all_levels(dhcpc_t)
 
 fs_getattr_all_fs(dhcpc_t)
 fs_search_auto_mountpoints(dhcpc_t)
@@ -146,6 +147,8 @@ init_rw_utmp(dhcpc_t)
 logging_send_syslog_msg(dhcpc_t)
 
 miscfiles_read_localization(dhcpc_t)
+# read generic SSL certificates
+miscfiles_read_generic_certs(dhcpc_t)
 
 modutils_run_insmod(dhcpc_t, dhcpc_roles)
 
-- 
2.11.0

